internal/collector/processor/tail_sampling/processor.go

```diff
@@ -249,9 +251,11 @@ func (tsp *tailSamplingSpanProcessor) ProcessSpans(batch *agenttracepb.ExportTra
 						zap.Uint64("failCount", failCount),
 						zap.Error(err))
 				}
+				stats.Record(policyAndDests.ctx, statCountTracesSampled.M(int64(1)))
```

Comment:
This and the one below are not needed since the trace was already counted: here it is just deciding what to do with spans that are arriving after the decision.